changelog/fragments/1693403216-Surface-errors-during-Agent's-enroll.yaml

```
@@ -0,0 +1,32 @@
+# Kind can be one of:
+# - breaking-change: a change to previously-documented behavior
+# - deprecation: functionality that is being removed in a later release
+# - bug-fix: fixes a problem in a previous version
+# - enhancement: extends functionality but does not break or fix existing behavior
+# - feature: new functionality
+# - known-issue: problems that we are aware of in a given version
+# - security: impacts on the security of a product or a user’s deployment.
+# - upgrade: important information for someone upgrading from a prior version
+# - other: does not fit into any of the other categories
+kind: feature
```

I think this should probably be a `bug-fix`? Seems like it's a bug for `elastic-agent install --url=... --enrollment-token-...` to succeed if enrollment fails. 
```suggestion
kind: bug-fix
```